constraint/pkg/client/drivers/local/args.go

```diff
@@ -1,114 +1,136 @@
 package local
 
 import (
+	"fmt"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client/errors"
 	"github.com/open-policy-agent/frameworks/constraint/pkg/externaldata"
-	"github.com/open-policy-agent/frameworks/constraint/pkg/handler"
 	"github.com/open-policy-agent/opa/ast"
 	"github.com/open-policy-agent/opa/storage"
 	"github.com/open-policy-agent/opa/storage/inmem"
 	"github.com/open-policy-agent/opa/topdown/print"
 	opatypes "github.com/open-policy-agent/opa/types"
 )
 
-type Arg func(*Driver)
+type Arg func(*Driver) error
 
 func Defaults() Arg {
-	return func(d *Driver) {
-		if d.compiler == nil {
-			d.compiler = ast.NewCompiler()
-		}
-
-		if d.modules == nil {
-			d.modules = make(map[string]*ast.Module)
-		}
-		if d.templateHandler == nil {
-			d.templateHandler = make(map[string][]string)
-		}
+	return func(d *Driver) error {
 		if d.storage == nil {
 			d.storage = inmem.New()
 		}
 
-		if d.capabilities == nil {
-			d.capabilities = ast.CapabilitiesForThisVersion()
+		if d.compilers.capabilities == nil {
+			d.compilers.capabilities = ast.CapabilitiesForThisVersion()
+		}
+
+		if d.compilers.externs == nil {
+			for allowed := range validDataFields {
+				d.compilers.externs = append(d.compilers.externs, fmt.Sprintf("data.%s", allowed))
+			}
 		}
 
 		// adding external_data builtin otherwise capabilities get overridden
 		// if a capability, like http.send, is disabled
 		if d.providerCache != nil {
-			d.capabilities.Builtins = append(d.capabilities.Builtins, &ast.Builtin{
+			newBuiltin := &ast.Builtin{
 				Name: "external_data",
 				Decl: opatypes.NewFunction(opatypes.Args(opatypes.A), opatypes.A),
-			})
+			}
+			d.compilers.capabilities.Builtins = append(d.compilers.capabilities.Builtins, newBuiltin)
```

Comment:
It looks like `d.compilers.capabilities` could be nil unless the `DisableBuiltins` option is set?

Reply:
`d.compilers.capabilities` is defaulted in Defaults() if unset:

```
if d.compilers.capabilities == nil {
	d.compilers.capabilities = ast.CapabilitiesForThisVersion()
}
```